www/nginx/src/opnsense/service/templates/OPNsense/Nginx/http.conf

```
@@ -110,13 +110,20 @@ server {
 {%     endif %}
     ssl_certificate_key /usr/local/etc/nginx/key/{{ single_servername }}.key;
     ssl_certificate /usr/local/etc/nginx/key/{{ single_servername }}.pem;
-    ssl_protocols TLSv1.2 TLSv1.3;
+{%     if server.tls_protocols is defined and server.tls_protocols != '' %}
+    ssl_protocols {{ server.tls_protocols.replace(',', ' ') }};
+{%     endif %}
     ssl_dhparam /usr/local/etc/dh-parameters.4096;
-    ssl_ciphers 'ECDHE-ECDSA-CAMELLIA256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-CAMELLIA256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-ECDSA-CAMELLIA128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-CAMELLIA128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-CAMELLIA256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-CAMELLIA256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-ECDSA-CAMELLIA128-SHA256:ECDHE-RSA-AES128-SHA256';
+{%     if server.tls_ciphers is defined and server.tls_ciphers != '' %}
+    ssl_ciphers {{ server.tls_ciphers }};
+{%     endif %}
```

```suggestion
    ssl_ciphers {{ server.tls_ciphers }};
```

basically this if we have a model required default :)

---

@kulikov-a ^^^

---

@fichtner I agree that this is more logical for this type) it just seemed to me that I often met such a construction (including for required fields) and decided that this is a general type of construction with some additional protection (do not add an option if, as a result of some error, the value turns out to be empty). my bad. thanks for the explanation )

---

It might be nitpicking on my part here. In general terms the sanity padding doesn’t change functionality and can be good to have, but trusting in the model can make the templates a lot easier to maintain (read, adjust and extend) as they become more complex.

---

@fichtner sorry, not my day( now we can get empty `ssl_ciphers` directive? (field is not Required. default value is set for backward compatibility). maybe you meant `ssl_protocols` part?

---

totally my fault. the same principle applies here although there is no not-set-default required here with `tls_ciphers ` and you can unset it.

---

uh! thanks!!) (I should have been more attentive when read the suggestion)